core/store/src/trie/mod.rs

```diff
@@ -47,6 +55,33 @@ enum TrieNode {
     Extension(Vec<u8>, NodeHandle),
 }
 
+#[derive(Clone, Debug)]
+struct TrieNodeWithSize {
+    node: TrieNode,
+    memory_usage: u64,
+}
+
+impl TrieNodeWithSize {
+    fn from_raw(rc_node: RawTrieNodeWithSize) -> TrieNodeWithSize {
+        TrieNodeWithSize { node: TrieNode::new(rc_node.node), memory_usage: rc_node.memory_usage }
+    }
+
+    fn new(node: TrieNode, memory_usage: u64) -> TrieNodeWithSize {
+        TrieNodeWithSize { node, memory_usage }
+    }
+
+    fn memory_usage(&self) -> u64 {
+        self.memory_usage
+    }
```

Comment:
it is strictly memory usage? I mean could it be another way of weighting state which is the most relevant to us?
may it me `state_cost` or something like that?